amp/src/test/java/org/digijava/kernel/ampapi/endpoints/activity/validators/RequiredValidatorTest.java

```diff
@@ -193,9 +192,9 @@ private APIField fd(String required) {
 
     private APIField fdList(String required) {
         APIField fd = fd(FIELD_ALWAYS_REQUIRED);
-        fd.setElementType(String.class);
-        fd.setFieldType(FIELD_TYPE_LIST);
-        fd.setItemType(FIELD_TYPE_STRING);
+        fd.getApiType().setElementType(String.class);
+        fd.getApiType().setFieldType(FieldType.LIST);
+        fd.getApiType().setItemType(FieldType.LONG);
```

Comment:
How it is possible to have element type String but item type LONG? We should avoid such cases by deriving item type from element type is possible.

Reply:
This is preexisting issue. I can refactor to avoid it.

Reply:
I can see that item type has changed from string to long in 00d58ce43309cac8211c21b9924e2838017210a6. Or maybe I am misunderstanding the situation.

Reply:
There were no strong correlation between item and element type to make a test to fail. The `APIType` now won't allow setting item type explicitly, so it will always derive from element type.